testing/btest/spicy/file-analysis-data-in-concurrent.zeek

```diff
@@ -1,12 +1,23 @@
 # @TEST-REQUIRES: have-spicy
 #
 # @TEST-EXEC: spicyz -d -o test.hlto ssh.spicy ./ssh-cond.evt
-# @TEST-EXEC: zeek -r ${TRACES}/ssh/single-conn.trace test.hlto %INPUT Spicy::enable_print=T >output
+# @TEST-EXEC: zeek -r ${TRACES}/ssh/single-conn.trace test.hlto %INPUT Spicy::enable_print=T >output 2>&1
 # @TEST-EXEC: btest-diff output
 
-event zeek_init() {
-    Analyzer::register_for_port(Analyzer::ANALYZER_SPICY_SSH, 22/tcp);
-}
+module SSH;
+
+global i: count = 0;
+
+function get_file_handle(c: connection, is_orig: bool): string
+	{
+	return cat(c$uid, ++i);
+	}
+
+event zeek_init()
+	{
+	Analyzer::register_for_port(Analyzer::ANALYZER_SPICY_SSH, 22/tcp);
+	Files::register_protocol(Analyzer::ANALYZER_SSH, [$get_file_handle=SSH::get_file_handle]); # use tag of replaced analyzer
```

Comment:
It isn't really clear to me why we cannot use the tag of this analyzer (from just the line above; trying that gives me _some result_ as well); could you add an expanded comment block explaining why we need to use the tag of the replaced analyzer (probably copy-and-pasted to everywhere we have the EOL comment now)?

Reply:
What result do you get? When I tried it, the `get_file_handle` just wouldn't be called, and instead I'd get a warning that the `get_file_handle()` handler was missing.

The reason that we need the replaced analyzer's tag is the `spicy_mgr->tagForProtocolAnalyzer(..)` call in the code, the translation happens there. We could skip it but then existing scripts written for the original analyzer wouldn't work, which isn't desired. And we can't check for both tags because the 1st failure triggers that warning (and that's coming from scriptland).

Reply:
You are right, I see that warning too, but also seemingly valid file IDs, e.g., for `spicy/file-data-in-at-offset.zeek`

```patch
-FU6P513gQKLVP6iwmf
-FU6P513gQKLVP6iwmf
+XXXXXXXXXX.XXXXXX warning in /.../zeek/scripts/base/frameworks/files/main.zeek, line 523: get_file_handle() handler missing for Analyzer::ANALYZER_SSH
+FPFmBApS7zfa2mU78
+FPbnGx3L0aN5H4YLPa
```

With that it probably make sense to stick with your current approach. Is this something which could be fixed as part of #3573?

Reply:
> You are right, I see that warning too, but also seemingly valid file IDs

Ok, got it. That's because it falls back to the previous built-in scheme to be able to continue still. That also means that existing analyses will keep working as before, people will just start seeing that warning now until scriptland gets updated--which seems better than breaking code.

> With that it probably make sense to stick with your current approach. Is this something which could be fixed as part of https://github.com/zeek/zeek/issues/3573?

Not quite following here: the PR's behavior for what tag to use in scriptland is so that `replaces` works transparently, which #3573 aims for too in a different context. So we wouldn't change anything further that this PR introduces. Unless I'm missing your point?

Reply:
I think I got myself confused and don't know enough about what happens in scriptland elsewhere, but that this the tag we use here makes a difference at all tells me that something `replaces` doesn't completely replace the analyzer reachable under a tag (#3573 is about reaching an analyzer via its name which is probably a different but related aspect). I am closing this.